VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/ImplementEqualsAndGetHashCode/ImplementEqualsAndGetHashCodeCodeFix.cs

```diff
@@ -0,0 +1,128 @@
+{
+    [ExportCodeFixProvider(DiagnosticId.ImplementEqualsAndGetHashCode + "CF", LanguageNames.CSharp), Shared]
+    public class ImplementEqualsAndGetHashCodeCodeFix : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(ImplementEqualsAndGetHashCodeAnalyzer.Rule.Id);
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            var statement = root.FindNode(diagnosticSpan);
+
+            context.RegisterCodeFix(
+                CodeAction.Create(string.Format(VSDiagnosticsResources.ImplementEqualsAndGetHashCodeCodeFixTitle),
+                    x => ImplementEqualsAndGetHashCodeAsync(context.Document, root, statement),
+                    ImplementEqualsAndGetHashCodeAnalyzer.Rule.Id),
+                diagnostic);
+        }
+
+        private Task<Document> ImplementEqualsAndGetHashCodeAsync(Document document, SyntaxNode root, SyntaxNode statement)
+        {
+            var classDeclaration = (ClassDeclarationSyntax)statement;
+
+            var newRoot = root.ReplaceNode(classDeclaration,
+                classDeclaration.AddMembers(GetEqualsMethod(classDeclaration), GetGetHashCodeMethod(classDeclaration)));
+            return Task.FromResult(document.WithSyntaxRoot(newRoot));
+        }
+
+        private MethodDeclarationSyntax GetEqualsMethod(ClassDeclarationSyntax classDeclaration)
+        {
+            var publicModifier = SyntaxFactory.Token(SyntaxKind.PublicKeyword);
+            var overrideModifier = SyntaxFactory.Token(SyntaxKind.OverrideKeyword);
+
+            var parameter = SyntaxFactory.Parameter(SyntaxFactory.Identifier("obj"))
+                .WithType(SyntaxFactory.ParseTypeName("object"));
+
+            var ifStatement =
+                SyntaxFactory.IfStatement(SyntaxFactory.ParseExpression($"obj == null || typeof({classDeclaration.Identifier}) != obj.GetType()"),
+                    SyntaxFactory.Block(SyntaxFactory.ParseStatement("return false;")));
+
+            var castStatement = SyntaxFactory.ParseStatement(
+                    $"var value = ({classDeclaration.Identifier}) obj;{Environment.NewLine}");
+
+            var fieldAndPropertyEqualityStatements = new List<string>();
+            foreach (var member in classDeclaration.Members)
+            {
+                if (member.IsKind(SyntaxKind.FieldDeclaration))
+                {
+                    var field = (FieldDeclarationSyntax)member;
+                    fieldAndPropertyEqualityStatements.AddRange(field.Declaration.Variables.Select(
```

Comment:
Can you add a unit test that uses chained field declarators? `private int _field1 = 5, _field2 = 10;` just to have this logic under test